component.yaml

```
@@ -0,0 +1,70 @@
+platform: Common
+buildPlatform: ReactNative
+displayName: Appcircle CodePush
+description: "Release a React Native update to Appcircle CodePush."
+inputs:
+- key: "AC_REPOSITORY_DIR"
+  defaultValue: "$AC_REPOSITORY_DIR"
+  isRequired: true
+  title: Repository Directory
+  description: "Relative path of the React Native project."
+  helpText:
+- key: "AC_CODE_PUSH_TOKEN"
+  defaultValue: "$AC_CODE_PUSH_TOKEN"
+  isRequired: true
+  title: Appcircle Personal Access Token
+  description: "Appcircle personal access token. You can create a new token in the Appcircle dashboard under `Settings > Security > Personal Access Token`."
+  helpText:
+- key: "AC_CODE_PUSH_SERVER_URL"
+  defaultValue: "$AC_CODE_PUSH_SERVER_URL"
+  isRequired: false
+  title: Appcircle CodePush Server URL
+  description: "Appcircle CodePush server URL. Leave it empty if you do not use self-hosted server."
+  helpText:
+- key: "AC_CODE_PUSH_AUTH_URL"
+  defaultValue: "$AC_CODE_PUSH_AUTH_URL"
+  isRequired: false
+  title: Appcircle CodePush Auth URL
+  description: "Appcircle CodePush auth URL. Leave it empty if you do not use self-hosted server."
+  helpText:
+- key: "AC_CODE_PUSH_APP_NAME"
+  defaultValue: "$AC_CODE_PUSH_APP_NAME"
+  isRequired: true
+  title: App Name
+  description: "This parameter specifies the name of the app in Appcircle. For example `MyApp-Android` or `MyApp-iOS`."
+  helpText:
+- key: "AC_CODE_PUSH_DEPLOYMENT_NAME"
+  defaultValue: "$AC_CODE_PUSH_DEPLOYMENT_NAME"
+  isRequired: true
+  title: Deployment Name
+  description: "This parameter specifies which deployment you want to release the update to. For example `Staging`."
+  helpText:
+- key: "AC_CODE_PUSH_TARGET_BINARY_VERSION"
+  defaultValue: "$AC_CODE_PUSH_TARGET_BINARY_VERSION"
+  isRequired: true
+  title: CodePush Target Binary Version
+  description: "This parameter specifies the target binary version of the app that this update is intended for. It should be in the format `1.0.0`."
+  helpText:
+- key: "AC_CODE_PUSH_DESCRIPTION"
+  defaultValue: "$AC_CODE_PUSH_DESCRIPTION"
+  isRequired: false
+  title: Description
+  description: "This parameter provides an optional change log for the deployment."
+  helpText:
+- key: "AC_CODE_PUSH_ROLLOUT_PERCENTAGE"
+  defaultValue: "$AC_CODE_PUSH_ROLLOUT_PERCENTAGE"
+  isRequired: false
+  title: Rollout Percentage
+  description: "This parameter specifies the percentage of users (as an integer between 1 and 100) that should be eligible to receive this update."
+  helpText:
+- key: "AC_CODE_PUSH_EXTRA_ARGUMENTS"
+  defaultValue: "$AC_CODE_PUSH_EXTRA_ARGUMENTS"
+  isRequired: false
+  title: Extra arguments
+  description: "Extra command line arguments for Appcircle. For example, add `--debug` for verbose logs."
```

If they know which command, they also know which arguments they can use:

```suggestion
  description: "Extra command line arguments for Appcircle CodePush CLI command. For example, add `--debug` for verbose logs."
```